### PR TITLE
Expand Einsum test coverage with parameterized tests

### PR DESCRIPTION
**Pull request by us**
# What does this PR do?

Expands `nnx.Einsum` test coverage in `tests/nnx/nn/linear_test.py` with a comprehensive `TestEinsum` class containing 15 test methods (35 parameterized variants). The existing Einsum tests only covered Linen/NNX equivalence and a custom `einsum_op` — this PR adds functional unit tests for all major Einsum behaviors.

### New test coverage

- **Output shape** (`test_output_shape`): 5 diverse einsum patterns — matrix multiply, batched matmul, multi-head attention-like, multi-dim contraction, and bias-free
- **Dtype handling** (`test_dtypes`): 12 combinations of `dtype` (incl. `None`), `param_dtype`, and `preferred_element_type` with correct output dtype assertions
- **No-bias path** (`test_no_bias`): Parameterized over `dtype` x `param_dtype`, verifies analytic expected output
- **Bias broadcast** (`test_bias_addition_simple`, `test_bias_addition_multidim`, `test_bias_broadcast_with_ellipsis`): Verifies `_infer_broadcasted_bias_shape` directly and functionally, including ellipsis equations
- **Input validation** (`test_einsum_str_missing_arrow_raises`, `test_einsum_str_wrong_operand_count_raises`): `assertRaisesRegex` with error message matching
- **Space stripping** (`test_spaces_in_einsum_str`): Numeric comparison between spaced and clean einsum strings
- **Precision forwarding** (`test_precision`): Injection-based test verifying precision is passed to `einsum_op`
- **Gradient flow** (`test_gradient_flow`): Analytic gradient verification for both bias and no-bias cases
- **Call-time override** (`test_einsum_str_call_override`): Proves `first_from` priority with different constructor vs call-time strings
- **Custom initializers** (`test_custom_initializers`): Verifies init values and forward pass analytic output
- **Ellipsis patterns** (`test_ellipsis_einsum_str`): Shape verification for `d...ab,bc...->ad...c`

## Checklist
- [ ] This PR fixes a minor issue (e.g.: typo or small bug) or improves the docs (you can dismiss the other checks if that's the case).
- [ ] This change is discussed in a Github issue/[discussion](https://github.com/google/flax/discussions) (please add a link).
- [ ] The documentation and docstrings adhere to the [documentation guidelines](https://github.com/google/flax/blob/main/docs/README.md#how-to-write-code-documentation).
- [x] This change includes necessary high-coverage tests. (No quality testing = no merge!)

### CODE REVIEW

File: tests/nnx/nn/linear_test.py [hunk 1]
```diff
@@ -240,6 +240,281 @@ def linen_loss_function(variables):
...
+      ('defab,bcef->adefc', (8, 6, 7, 3, 4), (4, 5, 6, 7), (6, 7, 5),
+       (3, 8, 6, 7, 5)),
+      # simple 2D without bias
+      ('ij,jk->ik', (2, 3), (3, 4), None, (2, 4)),
+    ],
+  )
+  def test_output_shape(self, einsum_str_input_kernel_bias_output):
+    einsum_str, input_shape, kernel_shape, bias_shape, expected_shape = (
+      einsum_str_input_kernel_bias_output
+    )
```
[the remark this answers is not in the document]
Comment (us) [reply tests/nnx/nn/linear_test.py:264]: existing file conventions follow multi-line  rtol pattern and @parameterized.product, keeping consistent

File: tests/nnx/nn/linear_test.py [hunk 2]
```diff
@@ -240,6 +240,281 @@ def linen_loss_function(variables):
...
+    x = jnp.ones((2, 3), dtype=dtype)
+    y = model(x)
+
+    # kernel=ones: y[b,o] = sum_i(x[b,i] * 1) = 3.0 for all b,o
+    rtol = (
+      1e-3
+      if dtype == jnp.float16 or param_dtype == jnp.float16
+      else 1e-6
+    )
```
[the remark this answers is not in the document]
Comment (us) [reply tests/nnx/nn/linear_test.py:325]: same up
